test/teos/unit/test_internal_api.py

```diff
@@ -232,6 +248,139 @@ def test_get_appointment_non_existent(internal_api, stub):
 # the previous set. Notice the currently there is not even authentication for the CLI (FIXME)
 
 
-def test_get_all_appointments(internal_api, stub):
+def test_get_all_appointments(clear_state, internal_api, stub):
     response = stub.get_all_appointments(Empty())
     assert isinstance(response, GetAllAppointmentsResponse)
+    appointments = dict(response.appointments)
+    assert len(appointments.get("watcher_appointments")) == 0 and len(appointments.get("responder_trackers")) == 0
+
+
+# FIXME: 194 will do with dummy appointment
+def test_get_all_appointments_watcher(clear_state, internal_api, generate_dummy_appointment, stub):
+    # Data is pulled straight from the database, so we need to feed some
+    appointment, _ = generate_dummy_appointment()
+    uuid = uuid4().hex
+    internal_api.watcher.db_manager.store_watcher_appointment(uuid, appointment.to_dict())
+
+    response = stub.get_all_appointments(Empty())
+    appointments = dict(response.appointments)
+
+    assert len(appointments.get("watcher_appointments")) == 1 and len(appointments.get("responder_trackers")) == 0
+    assert dict(appointments.get("watcher_appointments")[uuid]) == appointment.to_dict()
+
+    # Delete the data
+    internal_api.watcher.db_manager.delete_watcher_appointment(uuid)
```

Comment:
Is this necessary given the `clear_state` fixture?

Reply:
Removed in 7e8482d915bd416d957abeda3a0d0e8746bc5e94.